libcloud/compute/drivers/digitalocean.py

```diff
@@ -116,30 +120,93 @@ def create_node(self, name, size, image, location, ex_ssh_key_ids=None,
 
     def destroy_node(self, node):
-        params = {'scrub_data': '1'}
-        res = self.connection.request('/droplets/%s/destroy/' % (node.id),
-                                      params=params)
-        return res.status == httplib.OK
+        res = self.connection.request('/v2/droplets/%s' % (node.id),
+                                      method='DELETE')
+        return res.status == httplib.NO_CONTENT
+
+    def get_image(self, image_id):
+        """
+        Get an image based on an image_id
+
+        @inherits: :class:`NodeDriver.get_image`
+
+        :param image_id: Image identifier
+        :type image_id: ``int``
+
+        :return: A NodeImage object
+        :rtype: :class:`NodeImage`
+        """
+        res = self.connection.request('/v2/images/%s' % (image_id))
+        data = res.object['image']
+        return self._to_image(data)
+
+    def create_image(self, node, name):
+        """
+        Create an image fron a Node.
+
+        @inherits: :class:`NodeDriver.create_image`
+
+        :param node: Node to use as base for image
+        :type node: :class:`Node`
+
+        :param node: Name for image
+        :type node: ``str``
+
+        :rtype: ``bool``
+        """
+        params = {'type': 'snapshot', 'name': name}
+        res = self.connection.request('/v2/droplets/%s/actions' % (node.id),
+                                      params=params, method='POST')
+        return res.status == httplib.CREATED
+
+    def delete_image(self, image):
+        """Delete an image for node.
+
+        @inherits: :class:`NodeDriver.delete_image`
+
+        :param      image: the image to be deleted
+        :type       image: :class:`NodeImage`
+
+        :rtype: ``bool``
+        """
+        res = self.connection.request('/v2/images/%s' % (image.id),
+                                      method='DELETE')
+        return res.status == httplib.NO_CONTENT
 
     def ex_rename_node(self, node, name):
-        params = {'name': name}
-        res = self.connection.request('/droplets/%s/rename/' % (node.id),
-                                      params=params)
-        return res.status == httplib.OK
+        params = {'type': 'rename', 'name': name}
+        res = self.connection.request('/v2/droplets/%s/actions' % (node.id),
+                                      params=params, method='POST')
+        return res.status == httplib.CREATED
+
+    def ex_shutdown_node(self, node):
+        params = {'type': 'shutdown'}
+        res = self.connection.request('/v2/droplets/%s/actions' % (node.id),
+                                      params=params, method='POST')
+        return res.status == httplib.CREATED
+
+    def ex_power_on_node(self, node):
+        params = {'type': 'power_on'}
+        res = self.connection.request('/v2/droplets/%s/actions' % (node.id),
+                                      params=params, method='POST')
+        return res.status == httplib.CREATED
 
     def ex_list_ssh_keys(self):
```

Comment:
SSH key pair management is now part of the base API. So you should have the list_key_pairs(), get_key_pair(), create_key_pair(), delete_key_pair(). Also use the KeyPair() object:
    from libcloud.compute.base import KeyPair